src/main/java/org/osc/manager/rest/server/api/DomainApis.java

```
@@ -80,36 +147,268 @@ public PolicyEntity updatePolicy(@PathParam("domainid") Long domainid, @PathPara
+                        .createQuery(ApplianceManagerConnectorEntity.class);
+                Root<ApplianceManagerConnectorEntity> r = q.from(ApplianceManagerConnectorEntity.class);
+                q.select(r)
+                        .where(cb.and(cb.equal(r.get("name"), entity.getapplianceManagerConnector().getName())));
+
+                List<ApplianceManagerConnectorEntity> result = em.createQuery(q).getResultList();
+                if (result.isEmpty() == false) {
+                    domainElement.setapplianceManagerConnector(result.get(0));
+                    domainElement.setName(entity.getName());
+                    em.persist(domainElement);
+                    return entity;
+                } else {
+                    return null;
+                }
+            }
+        });
+    }
+
+    /**
+     * Updates the Domain for a given domain Id
+     *
+     * @return - updated Domain
+     */
+    @Path("/{domainid}")
+    @PUT
+    public DomainEntity updateDomain(@PathParam("domainid") Long domainid, DomainEntity entity) {
+
+        logger.info("Updating Domain Entity ID...:" + domainid);
+
+        return this.txControl.required(new Callable<DomainEntity>() {
+
+            @Override
+            public DomainEntity call() throws Exception {
+
+                CriteriaBuilder cb = em.getCriteriaBuilder();
+                CriteriaQuery<DomainEntity> q = cb.createQuery(DomainEntity.class);
+                Root<DomainEntity> r = q.from(DomainEntity.class);
+                q.select(r).where(cb.and(cb.equal(r.get("Id"), domainid)));
+                List<DomainEntity> result = em.createQuery(q).getResultList();
+                if (result.isEmpty() == false) {
+                    result.get(0).setName(entity.getName());
+                    em.persist(result.get(0));
+                    return entity;
+                }
+                else {
+                    return null;
```

identation is broken also this should fail with not found

---

error 404 shall be added in subsequent user stories update.

---

ok but for now at least ensure that the situation is logged and an exception is thrown (that will likely show up in fiddler as 500 which is not good but better than silently returning null.